jdbc-yugabytedb/src/main/java/org/postgresql/Driver.java

```diff
@@ -529,17 +539,19 @@ private static Connection getConnectionBalanced(LoadBalanceProperties lbprops) t
           }
           // TODO log exception go to the next one after adding to failed list
           LOGGER.log(Level.WARNING,
-            "couldn't connect to {0}, adding to failed host list", chosenHost);
+            "couldn't connect to " + chosenHost + ", adding it to failed host list");
           failedHosts.add(chosenHost);
           loadBalancer.updateFailedHosts(chosenHost);
-          if (connectionCreated) {
-            // ignore refresh. return the connection after updating map
-            loadBalancer.updateConnectionMap(chosenHost, 1);
```

Comment:
In the previous version, we were updating the `ConnectionMap` on successful connection, now the code is being modified to decrement the connection attempt on connection failure/unsuccessful. Where are we incrementing the `ConnectionMap` on successful connection creation? trying to understand the current implementation behavior.

Reply:
That is the main change. We are incrementing it in the getLeastLoadedServer method itself. And decrementing it only when the actual connection to it later fails.